File: problem010.pl
#!/usr/bin/perl

use strict;
use warnings;
my $num = 2000000;
my $sum = 0;

for(my $i = $num; $i > 1; $i--){
    
    my $notPrime = 0;
    for(my $j = 2; $j < $i; $j++){
        if($i % $j == 0){
            $notPrime = 1;
            last;
        }
    }

    if(!$notPrime){
        $sum += $i;
    }

    print "$i\n";
}

print "Answer: $sum\n";

File: problem010.cpp
#include <iostream>
#include <vector>
int main(){
    int sum = 0;
    for(int i = 2000000; i > 1; i--){
        int notPrime = 0;
        for(int j = 2; j < i; j++){
            if(i % j == 0){
                notPrime = 1;
                break;
            }
        }

        if(notPrime == 0){
            sum += i;
        }

        if(i % 1000 == 0){
            std::cout << i << std::endl;
        }
    }
    std::cout << "Answer: " << sum << std::endl; 
    return 0;
}
